blog/views.py

```diff
@@ -0,0 +1,49 @@
+from django.views import generic
+from django.shortcuts import render, get_object_or_404
+from django.shortcuts import redirect
+
+from blog.models import Post, Commentary
+
+
+class PostListView(generic.ListView):
+    model = Post
+    template_name = "blog/index.html"
+    queryset = Post.objects.order_by("-created_time").select_related("owner")
+    paginate_by = 5
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
+
+
+class CommentaryCreateView(generic.View):
+    @staticmethod
+    def post(request, pk):
```

Comment:
The `CommentaryCreateView` should ideally use a form to handle POST requests instead of manually processing `request.POST`. Consider using Django's `FormView` or `CreateView` for better form handling and validation.